cert-manager: add trusted internal ca when configured

infra-monkey opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
In order to use an acme issuer with an internal authority, the CA of the authority must be trusted by cert-manager.
If not, the issuer with throw errors saying the certificate of the acme uri is unknown.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:
I tried to keep it simple.
One additional variable in the addon inventory to specify the trusted internal ca.
A configmap is create with the content of the ca.
This configmap is mounted in the cert-manager container in /etc/ssl/certs in order to be trusted.
As not everyone will use acme with an internal authority, the cm and mount are not defined if the variable is not defined.

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Add a new option `cert_manager_trusted_internal_ca` to specify trusted internal ca of cert_manager.
```

oomichi commented:
@infra-monkey Thank you so much for your explanation, I got it.
This additional option doesn't break the existing features by default.
So I feel it is fine to add it and get feedback if any problem.

/ok-to-test
/lgtm

infra-monkey commented:
@oomichi thank you for the approval.
Indeed my initial description was light.

I put "none" for the user facing change. As there is a new option in the inventory, should I specifiy it there?

oomichi commented:
> 
> 
> @oomichi thank you for the approval. Indeed my initial description was light.
> 
> I put "none" for the user facing change. As there is a new option in the inventory, should I specifiy it there?

That is a nice point.
It would be better to write a release note instead of NONE like
```
Add a new option cert_manager_trusted_internal_ca to specify internal ca of cert_manager
```
or something.

infra-monkey commented:
@oomichi a vagrant job failed in the pipeline. How can I restart the tests?

cristicalin commented:
@infra-monkey  would you mid updating https://github.com/kubernetes-sigs/kubespray/blob/master/docs/cert_manager.md with some details on the new variable and the intended purpose?

infra-monkey commented:
> @infra-monkey would you mid updating https://github.com/kubernetes-sigs/kubespray/blob/master/docs/cert_manager.md with some details on the new variable and the intended purpose?

Great point. I'll do that. I hope my english will be good enough.

cristicalin commented:
It might be a good idea to rebase your branch on latest master since the CI job is complaining about failing to merge your doc changes.

infra-monkey commented:
@cristicalin Rebased from master. It bring a lot of changes from master that have nothing to do with this change.
Sorry I'm not a git guru.

CI fails, but because it can't pull containers.

cristicalin commented:
The relevant failure on the CI is this:

```
Using index info to reconstruct a base tree...
M	docs/cert_manager.md
Falling back to patching base and 3-way merge...
Auto-merging docs/cert_manager.md
CONFLICT (content): Merge conflict in docs/cert_manager.md
Patch failed at 0004 Update documentation
Use 'git am --show-current-patch' to see the failed patch
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
error: Failed to merge in the changes.
Uploading artifacts for failed job 00:00
Uploading artifacts...
WARNING: cluster-dump/: no matching files          
ERROR: No files to upload                          
```

https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/1735187854#L40

infra-monkey commented:
@cristicalin I've cleaned up the commits and fixed the merge.
Now matrix-ci fails https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/1741025781
Looks like a pip issue

cristicalin commented:
Unfortunately it seems our CI python was pushed to 3.10 and is now breaking in interesting ways. I pushed a fix https://github.com/kubernetes-sigs/kubespray/pull/8153 but there may be other breakage.

infra-monkey commented:
> Unfortunately it seems our CI python was pushed to 3.10 and is now breaking in interesting ways. I pushed a fix #8153 but there may be other breakage.

I just saw that. I'll wait for you PR to be merged in master and include it here.
Thank you for looking into it

floryut commented:
> > Unfortunately it seems our CI python was pushed to 3.10 and is now breaking in interesting ways. I pushed a fix #8153 but there may be other breakage.
> 
> I just saw that. I'll wait for you PR to be merged in master and include it here. Thank you for looking into it

Merged now you can rebase 👍

infra-monkey commented:
I cherry picked the CI fix.

infra-monkey commented:
@oomichi @cristicalin @floryut Seems the pipeline fails on a timeout downloading a file.
I don't see how my changes could influence this part.
How should I proceed?

floryut commented:
> @oomichi @cristicalin @floryut Seems the pipeline fails on a timeout downloading a file. I don't see how my changes could influence this part. How should I proceed?

yup there is an issue with calico release, I've submitted a PR and I'm trying to get in touch with Calico maintainers to understands what the f** happened

infra-monkey commented:
> yup there is an issue with calico release, I've submitted a PR and I'm trying to get in touch with Calico maintainers to understands what the f** happened
I have seen your PR. Indeed it is strange that the hash would change without a version change.
I'll wait for your go before rebasing the fix.
Bad timing for me but at least the ci failures are caught early

cristicalin commented:
@infra-monkey #8157 just merged, please rebase and lets try this run one more time.

infra-monkey commented:
YEAH ! It finally passed!

oomichi commented:
> 
> 
> YEAH ! It finally passed!

Current pull request is not rebased yet, it just contains necessary commits in the pull request.
Please rebase this pull request like
```
git checkout master
git remote add upstream https://github.com/kubernetes-sigs/kubespray.git
git fetch upstream
git merge upstream/master
git checkout <your branch>
git rebase master
git push -f origin <your branch>
```

infra-monkey commented:
@oomichi I didn't think you wanted a full rebase before merge. I misunderstood the request to rebase.
It is done now. CI running

cristicalin commented:
@infra-monkey it is the norm to have only your commits in a PR unless you are doing backports to older branches. This is needed to keep the git commit history clean and also to avoid issues when github squashes the commits in the final merge.

cristicalin commented:
/approve

floryut reviewed (approved):
@infra-monkey Thank you for the PR 👍

infra-monkey commented:
> @infra-monkey Thank you for the PR +1

Thank you all for the guidance and patience :)

cristicalin commented:
/lgtm
